eva-accession-ws/src/test/java/uk/ac/ebi/eva/accession/ws/VariantAccessioningRestControllerTest.java

```diff
@@ -47,46 +46,22 @@
 public class VariantAccessioningRestControllerTest {
 
     @Autowired
-    private TestRestTemplate testRestTemplate;
+    private BasicRestController basicRestController;
 
     @Autowired
-    private SubmittedVariantAccessioningRepository accessioningRepository;
-
-    @Test
-    public void testRestApi() {
-        String url = "/v1/variant";
-        HttpEntity<Object> requestEntity = new HttpEntity<>(getListOfVariantMessages());
-        ResponseEntity<Map> response = testRestTemplate.exchange(url, HttpMethod.POST, requestEntity, Map.class);
-        assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals(2, response.getBody().size());
-    }
+    private TestRestTemplate testRestTemplate;
 
-    @Test
-    public void requestPostTwiceAndWeGetSameAccessions() {
-        String url = "/v1/variant";
-        HttpEntity<Object> requestEntity = new HttpEntity<>(getListOfVariantMessages());
-        ResponseEntity<Map> response = testRestTemplate.exchange(url, HttpMethod.POST, requestEntity, Map.class);
-        assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals(2, response.getBody().size());
-        assertEquals(2, accessioningRepository.count());
+    private static final Long ACCESSION = 10000000001L;
 
-        //Accessing Post Request again with same files
-        response = testRestTemplate.exchange(url, HttpMethod.POST, requestEntity, Map.class);
-        assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals(2, response.getBody().size());
-        assertEquals(2, accessioningRepository.count());
-    }
+    private static final String URL = "/v1/submitted-variants/";
 
     @Test
-    public void testGetVariantsRestApi() {
-        String getAccessionsUrl = "/v1/variant";
-        HttpEntity<Object> requestEntity = new HttpEntity<>(getListOfVariantMessages());
-        ResponseEntity<Map> getAccessionsResponse = testRestTemplate.exchange(getAccessionsUrl, HttpMethod.POST,
-                requestEntity, Map.class);
-        assertEquals(HttpStatus.OK, getAccessionsResponse.getStatusCode());
-        assertEquals(2, getAccessionsResponse.getBody().size());
-        Object[] accessions = getAccessionsResponse.getBody().keySet().toArray();
-        String getVariantsUrl = "/v1/variant/" + accessions[0] + "," + accessions[1];
+    public void testGetVariantsRestApi() throws AccessionCouldNotBeGeneratedException {
+        Map<Long, SubmittedVariantDTO> generatedAccessions = basicRestController.generateAccessions(
+                getListOfVariantMessages());
+        assertEquals(2, generatedAccessions.size());
+        Object[] accessions = generatedAccessions.keySet().toArray();
```

Comment:
I don't  like a lot having an Object array here... 
What about something like this?

```
String accessions = generatedAccessions.keySet().stream().map(acc -> acc.toString()).collect(Collectors.joining(","));
String getVariantsUrl = URL + accessions;
```